core/src/main/java/org/elasticsearch/common/settings/Settings.java

```diff
@@ -718,6 +718,14 @@ public Builder setSecureSettings(SecureSettings secureSettings) {
             if (secureSettings.isLoaded() == false) {
                 throw new IllegalStateException("Secure settings must already be loaded");
             }
+            if (secureSettings.getSettingNames().isEmpty()) {
+                // TODO: fix this leniency!!!
```

Comment:
I ran into this a couple of times working on the branch. Can we throw an exception instead?